packages/database/ios/RNFBDatabase/RNFBDatabaseCommon.m

```diff
@@ -76,33 +87,47 @@ + (void)setDatabaseConfig:(FIRDatabase *)firDatabase
       NSInteger *cacheSizeBytes = [preferences getIntegerValue:DATABASE_PERSISTENCE_CACHE_SIZE defaultValue:(NSInteger *) 10000000];
       [firDatabase setPersistenceCacheSizeBytes:(NSUInteger) cacheSizeBytes];
     }
+      
+    @synchronized (configSettingsLock) {
+        configSettingsLock[lockKey] = @YES;
+    }
+      
   } @catch (NSException *exception) {
     if (![@"FIRDatabaseAlreadyInUse" isEqualToString:exception.name]) {
       @throw exception;
+    } else {
+      @synchronized (configSettingsLock) {
+      configSettingsLock[lockKey] = @YES;
+      }
     }
   }
 }
 
 + (FIRDatabaseReference *)getReferenceForDatabase
     :(FIRDatabase *)firebaseDatabase
                                              path:(NSString *)path {
-  return [firebaseDatabase referenceWithPath:path];
+    @synchronized (configSettingsLock) {
+      return [firebaseDatabase referenceWithPath:path];
+    }
 }
 
 + (FIRDatabaseReference *)getReferenceForDatabase
     :(NSString *)key
                                  firebaseDatabase:(FIRDatabase *)firebaseDatabase
                                              path:(NSString *)path {
-  FIRDatabaseReference *cachedReference = references[key];
+    @synchronized (configSettingsLock) {
+      FIRDatabaseReference *cachedReference = references[key];
 
-  if (cachedReference != nil) {
-    return cachedReference;
-  }
+    if (cachedReference != nil) {
+      return cachedReference;
+    }
 
-  FIRDatabaseReference *databaseReference = [firebaseDatabase referenceWithPath:path];
+    FIRDatabaseReference *databaseReference = [firebaseDatabase referenceWithPath:path];
 
-  references[key] = databaseReference;
-  return databaseReference;
+    references[key] = databaseReference;
+        
+    return databaseReference;
+    }
 }
```

Comment:
Formatting is off here

Reply:
The formatting is in line with the rest of the code indented within the methods. 5 columns from the left.

Reply:
Hm, this looks wrong though: 

![image](https://user-images.githubusercontent.com/842078/79477971-bf89f200-8002-11ea-914b-9157b120305f.png)

Reply:
yeah, just had another look, it's inline with the code above which I have formatted to be two spaces after beginning of method signature. but I didn't take into account the plus/minus sign at the beginning of methods. Will change.

Reply:
![image](https://user-images.githubusercontent.com/5347038/79565763-41ccf180-80a9-11ea-9ca2-1516c019978b.png)

Formatting is still off here 👀

Reply:
huh, it is now three columns from the left like all the other code within methods. I don't understand how it is rendering differently in github 🤔:
![Screenshot 2020-04-17 at 16 02 19](https://user-images.githubusercontent.com/16018629/79583410-fd4f4f00-80c4-11ea-8a4f-570bc4546c82.png)